crates/core/component/auction/src/component/dutch_auction.rs

```diff
@@ -184,6 +367,58 @@ pub(crate) trait DutchAuctionManager: StateWrite {
 
 impl<T: StateWrite + ?Sized> DutchAuctionManager for T {}
 
+#[async_trait]
+pub(crate) trait HandleDutchTriggers: StateWrite {
+    /// Process the trigger height for a [`DutchAuction`],
+    async fn process_trigger(&mut self, trigger_height: u64) -> Result<()> {
+        use futures::StreamExt;
+        // TODO(erwan): somewhat self-defeating, but this is a first pass.
+        let auction_ids: Vec<AuctionId> = self
+            .stream_dutch_ids_by_trigger(trigger_height)
+            .await
+            .collect()
+            .await;
```

Comment:
Hmm, assuming the self-defeating is about getting all the IDs and then processing them in sequence, I don't think this is avoidable, we don't allow concurrent updates to the DEX...